specs/APIReview_AllowExternalDrop.md

```diff
@@ -0,0 +1,100 @@
+# Background
+Currently the functionality of dragging and dropping external objects(e.g. files, hyperlinks) into webview2 is default enabled in webview2 and there is no way to disable it. Some developers may want to disbale this functionality in their applications based upon their scenarios. According to such feature requirements, we add the new API to provide developers with the capability to configure the external drag&drop functionality.
+
+# Description
+We add a new `AllowExternalDrop` property in `CoreWebView2Controller`. 
+This API allows end developers to toggle the external drag&drop functionality easily.
+If it's disabled, drag&drop objects from outside into current webview2 will be forbidden.
+To be more specific, some behaviors listed below will be impacted by the toggle of this property.
+
+* Drag&Drop files on disk to webview2  
+* Drag&Drop hyperlinks from browser to webview2
+* Drag&Drop hyperlinks from one webview2 to another webview2
+
+Some behaviors are not impacted by the toggle of this property like drag&drop selected text to webview2. That means toggle of this property has no impact on the expected behavior of this action.
+
+By default, AllowExternalDrop is enabled to keep consistent with the behavior we had before the API is added.
+
+Please note that drag&drop anything from webview2 to external(outside current webview2) will not be impacted by this property.
+
+# Examples
+## C++
+
+```cpp
+wil::com_ptr<ICoreWebView2Controller4> m_controller4;
+void ToggleAllowDrop()
```

Comment:
These samples don't really demonstrate anything related to the drag/drop functionality. You could search/replace "AllowDrop" with "DoTheHokeyPokey" and it still would be a sample.

I expect something more like
```cpp
// This hypothetical app allows dropping external content
// only on the AttachReceipts page.
HRESULT OnNavigationStarting(ICoreWebView2* sender, ICoreWebView2NavigationStartingEventArgs* args)
{
    // Disable external drop while navigating.
    if (auto controller4 = m_controller.try_query<ICoreWebView2Controller4>(); controller4)
    {
        CHECK_FAILURE(controller4->put_AllowExternalDrop(FALSE));
    }
    // Decide whether to enable external drop when we finish navigating.
    wil::unique_cotaskmem_string uri;
    CHECK_FAILURE(args->get_Uri(&uri));
    m_allowExternalDropOnNavigationComplete =  (uri_equal(uri.get(), L"myapp://AttachReceipts.html"));
}

HRESULT OnNavigationCompleted(ICoreWebView2* sender, ICoreWebView2NavigationCompletedEventArgs* args)
{
    if (auto controller4 = m_controller.try_query<ICoreWebView2Controller4>(); controller4)
    {
        CHECK_FAILURE(controller4->put_AllowExternalDrop(m_allowExternalDropOnNavigationComplete));
    }
}
```

Reply:
I was imagining this was a property you'd mostly set once for a WebView2 and so didn't have an interesting idea for a sample, but I like this sample.

Reply:
This does raise the question is when changes to the property take effect.

1. Immediately.
2. At the next navigation.
3. At the creation of a WebView from the controller.

My proposed sample assumes 1.

If 3, then the drag/drop setting is not meaningfully settable once the WebView is created, so disabling it would have to be done before creating the webview.

If 2, then changing the setting at navigation start is too late. Not sure how to fix that.

Reply:
My implementation is #1. It will take effect immediately after users set the property, including set the property at the creation of WebView.

Reply:
Good suggestion about the sample. I changed them to the proposed ones which are more straightforward and friendly to demonstrate this API. Thanks @oldnewthing.